source/faas-automated-testing/faas-automated-testing.rst

```diff
@@ -0,0 +1,303 @@
+======================
+FaaS Automated Testing
+======================
+
+:Status: 
+:Minimum Server Version: 3.6
+
+.. contents::
+
+--------
+
+Abstract
+========
+
+This specification is about the ability for drivers to automate tests for
+"Functions as a Service" from continuous integration.
+
+META
+====
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`_.
+
+Specification
+=============
+
+Terms
+-----
+
+FaaS
+~~~~
+
+"Function as a Service", such as AWS Lambda.
+
+Implementing Automated FaaS Tests
+---------------------------------
+
+AWS Lambda
+~~~~~~~~~~
+
+This section describes the required setup of an AWS Lambda function and the
+steps needed to automate the deployment and execution of the function in
+Evergreen.
+
+Local Execution
+***************
+
+Prerequisites
+`````````````
+
+For the initial local setup the following are required:
+
+- The docker daemon running on the local machine.
+- The [https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/install-sam-cli.html](AWS SAM CLI)
+
+The following environment variables MUST be present (See DRIVERS-2384 google doc for values):
```

Comment:
```suggestion
The following environment variables MUST be present:
```

Details about where credentials are available should not be contained in the spec, as this information might change in the future and shouldn't require a spec change. The environment variables here are also required for a number of other tests, so there's a good chance drivers will already have them set in their CI environment.

Reply:
I've removed the wording now.